test/spec/fixtures/schema.sql

```diff
@@ -3368,3 +3368,72 @@ $$ language sql;
 create function returns_setof_record_params(id int, name text) returns setof record as $$
 select * from projects p where p.id >= $1 and p.name like $2;
 $$ language sql;
+
+create function raise_sqlstate_test1() returns void
+  language plpgsql
+  as $$
+begin
+    raise sqlstate 'PGRST' USING
+      message = '{"code":"123","message":"ABC","details":"DEF","hint":"XYZ"}',
+      detail = '{"status":332,"description":"My Custom Status","headers":{"X-Header":"str"}}';
```

Comment:
@taimoorzaeem @laurenceisla `description` by itself feels a bit loose. The official term for the status description is "reason phrase". See:

- https://www.rfc-editor.org/rfc/rfc7230#section-3.1.2
- https://www.ibm.com/docs/en/cics-ts/5.4?topic=concepts-status-codes-reason-phrases

So I think there are two options for making this more correct:

1) One single field for the status line, the reason phrase being optional: `"status": "332"` or `"status": "332 My Custom Status"`. Harder to implement but better DX for the user.
2) Change `description` to `status_text` (also a known term for it). Simpler to implement but less DX for the user. Though the case where they'd want custom status codes is too rare, so it doesn't matter as much.

Personally I'd go with option 2.

Reply:
Cool! I'd choose option 2 too.